src/devices/machine/cs8900a.cpp

```diff
@@ -0,0 +1,1522 @@
+#define CS8900_PP_ADDR_END             0x1000 /* memory to CS8900_PP_ADDR_END-1 is used */
+
+
+/* CS8900_PP_ADDR_CONF_CTRL is subdivided: */
+#define CS8900_PP_ADDR_CC_RXCFG        0x0102 /* # RW - 4.4.6.,  p. 52 - 0003 */
+#define CS8900_PP_ADDR_CC_RXCTL        0x0104 /* # RW - 4.4.8.,  p. 54 - 0005 */
+#define CS8900_PP_ADDR_CC_TXCFG        0x0106 /*   RW - 4.4.9.,  p. 55 - 0007 */
+#define CS8900_PP_ADDR_CC_TXCMD        0x0108 /*   R- - 4.4.11., p. 57 - 0009 */
+#define CS8900_PP_ADDR_CC_BUFCFG       0x010A /*   RW - 4.4.12., p. 58 - 000B */
+#define CS8900_PP_ADDR_CC_LINECTL      0x0112 /* # RW - 4.4.16., p. 62 - 0013 */
+#define CS8900_PP_ADDR_CC_SELFCTL      0x0114 /*   RW - 4.4.18., p. 64 - 0015 */
+#define CS8900_PP_ADDR_CC_BUSCTL       0x0116 /*   RW - 4.4.20., p. 66 - 0017 */
+#define CS8900_PP_ADDR_CC_TESTCTL      0x0118 /*   RW - 4.4.22., p. 68 - 0019 */
+
+/* CS8900_PP_ADDR_STATUS_EVENT is subdivided: */
+#define CS8900_PP_ADDR_SE_ISQ          0x0120 /*   R- - 4.4.5.,  p. 51 - 0000 */
+#define CS8900_PP_ADDR_SE_RXEVENT      0x0124 /* # R- - 4.4.7.,  p. 53 - 0004 */
+#define CS8900_PP_ADDR_SE_TXEVENT      0x0128 /*   R- - 4.4.10., p. 56 - 0008 */
+#define CS8900_PP_ADDR_SE_BUFEVENT     0x012C /*   R- - 4.4.13., p. 59 - 000C */
+#define CS8900_PP_ADDR_SE_RXMISS       0x0130 /*   R- - 4.4.14., p. 60 - 0010 */
+#define CS8900_PP_ADDR_SE_TXCOL        0x0132 /*   R- - 4.4.15., p. 61 - 0012 */
+#define CS8900_PP_ADDR_SE_LINEST       0x0134 /*   R- - 4.4.17., p. 63 - 0014 */
+#define CS8900_PP_ADDR_SE_SELFST       0x0136 /*   R- - 4.4.19., p. 65 - 0016 */
+#define CS8900_PP_ADDR_SE_BUSST        0x0138 /* # R- - 4.4.21., p. 67 - 0018 */
+#define CS8900_PP_ADDR_SE_TDR          0x013C /*   R- - 4.4.23., p. 69 - 001C */
+
+#define MAX_TXLENGTH 1518
+#define MIN_TXLENGTH 4
+
+#define MAX_RXLENGTH 1518
+#define MIN_RXLENGTH 64
+
+#define CS8900_TX_IDLE 0
+#define CS8900_TX_GOT_CMD 1
+#define CS8900_TX_GOT_LEN 2
+#define CS8900_TX_READ_BUSST 3
+
+#define CS8900_RX_IDLE 0
+#define CS8900_RX_GOT_FRAME 1
+
+#ifdef CS8900_DEBUG_FRAMES
+    #define return( _x_ ) \
+    { \
+        int retval = _x_; \
+        \
+        log_message(cs8900_log, "%s correct_mac=%u, broadcast=%u, multicast=%u, hashed=%u, hash_index=%u", (retval? "+++ ACCEPTED":"--- rejected"), *pcorrect_mac, *pbroadcast, *pmulticast, *phashed, *phash_index); \
+        \
+        return retval; \
+    }
+#endif
+
+#define PP_PTR_AUTO_INCR_FLAG 0x8000 /* auto increment flag in package pointer */
+#define PP_PTR_FLAG_MASK      0xf000 /* is always : x y 1 1 (with x=auto incr) */
+#define PP_PTR_ADDR_MASK      0x0fff /* address portion of packet page pointer */
+
+
+#define LO_BYTE(x)      (u8)((x) & 0xff)
+#define HI_BYTE(x)      (u8)(((x) >> 8) & 0xff)
+#define LOHI_WORD(x,y)  ( (u16)(x) | ( ((u16)(y)) <<8 ) )
+
+//------------------- END #defines ---------------------
+
+
+unsigned long cs8900a_device::crc32_buf(const char *buffer, unsigned int len)
```

Comment:
I suggest you can use util::crc32_creator::simple() here instead of another implementation.